info-v4-object.go

```diff
@@ -0,0 +1,145 @@
+//
+// Copyright (c) 2015-2025 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+package madmin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// ObjectInfoOptions ...
+type ObjectInfoOptions struct {
+	Bucket, Object string
+}
+
+// ObjectInfo calls minio to search for all files and parts
+// related to the given object, across all disks.
+func (adm *AdminClient) ObjectInfo(ctx context.Context, objOpts ObjectInfoOptions) (files map[string]*ObjectInspectInfo, err error) {
+	form := make(url.Values)
+	form.Add("bucket", objOpts.Bucket)
+	form.Add("object", objOpts.Object)
+
+	resp, err := adm.executeMethod(ctx,
+		http.MethodGet,
+		requestData{
+			relPath:     fmt.Sprintf(adminAPIPrefixV4 + "/object-info"),
+			queryValues: form,
+		})
+	defer closeResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, httpRespToErrorResponse(resp)
+	}
+
+	bb, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	files = make(map[string]*ObjectInspectInfo)
+	err = json.Unmarshal(bb, &files)
```

Comment:
```suggestion
	dec := json.NewDecoder(resp.Body)
	err = dec.Decode(&files)
```

Reply:
Why do we need a decoder ? it's not a streaming response